### DISCUSSION TITLE
Question: How to set auto deleting messages for a chat ?

### DISCUSSION THREAD

**Discussion by maxpayne3d** [python-telegram-bot/python-telegram-bot#2864] [Q&A] [answered]
I can't find a way to set message auto delete for a chat. Can anyone help ?
Thanks.

**Comment by harshil21** [comment, marked as answer]
You can use PTB's [Job Queue](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Extensions-%E2%80%93-JobQueue) for this. 

A tiny snippet which would achieve what you want:
``` python
msg = bot.send_message(...)
context.job_queue.run_once(lambda c: c.job.context(), 60, context=msg.delete)
```
This would delete the message after 60 seconds.

**Reply by Bibo-Joshi** [reply]
nice, never thought about passing a callable as `context` :D

**Reply by Poolitzer** [reply]
I would like to get a mind blown emoji reaction for this.

**Reply by Bibo-Joshi** [reply]
🤯

**Reply by maxpayne3d** [reply]
Thank you for the answer. Seems like a bit over my experience but I will look into it.
